GreeterTest.java

```diff
@@ -0,0 +1,51 @@
+import static org.junit.Assert.assertEquals;
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.After;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+/*
+ * thanks to dfa at  http://stackoverflow.com/a/1119559
+ */
+public class GreeterTest{
+    PrintStream origOut;
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+
+    @Before
+    public void setUpStream(){
+        origOut = System.out;
+        System.setOut(new PrintStream(outContent));
+    }
+    @After
+    public void cleanUpStream(){
+        System.setOut(origOut);
+    }
+
+    @Test
+    public void greetDefault(){
+        Greeter greeter = new Greeter();
+        greeter.parseArgs();
```

Comment:
parseArgs should be called from within Greeter main.  The command line args are coming from the user invoking Greeter.